src/blueapi/service/handler.py

```diff
@@ -80,7 +80,8 @@ def _publish_event_stream(self, stream: EventStream, destination: str) -> None:
 
     def stop(self) -> None:
         self.worker.stop()
-        self.messaging_template.disconnect()
+        if self.messaging_template.is_connected():
+            self.messaging_template.disconnect()
```

Comment:
Hmm, I definitely intended `MessagingTemplate.disconnect` to be idempotent. If it's not that's probably a bug in itself, worth raising an issue over

Reply:
This was pretty much the crux of the issue. It was trying to disconnect something that was already disconnected and was raising exceptions.

Reply:
https://github.com/DiamondLightSource/blueapi/issues/340

Reply:
Hmm, I _think_ I'd prefer it be made to only disconnect if connected, similar to this. What do you think?